code/game/machinery/bots/draculabot.dm

```diff
@@ -0,0 +1,228 @@
+//Rewards dispensed once every 55u donated.
+
+/obj/machinery/bot/bloodbot
+	name = "Doctor Acula"
+	desc = "A blood donation medibot. If wearing a cape, notify security immediately."
+	icon = 'icons/obj/aibots.dmi'
+	icon_state = "bloodbot0"
+	density = 0
+	anchored = 0
+	health = 20
+	maxhealth = 20
+	req_access =list(access_medical)
+	var/mob/living/carbon/human/target //Only used if emagged
+	var/currently_drawing_blood = 0 //One patient at a time.
+	var/quiet = 0
+	var/since_last_reward = 0 //Once every 55u, dispense reward
+	var/list/possible_rewards = list(/obj/item/weapon/reagent_containers/food/snacks/ijzerkoekje) //Could add more later or adminbus
+	var/reserve_bags = 3 //How many empty bags do we have?
+	bot_type = MED_BOT //This handles which HUD the bot talks to.
+
+/obj/machinery/bot/bloodbot/update_icon()
+	icon_state = "bloodbot[emagged][currently_drawing_blood]"
+
+/obj/machinery/bot/bloodbot/turn_on()
+	..()
+	update_icon()
+	updateUsrDialog()
+
+/obj/machinery/bot/bloodbot/turn_off()
+	..()
+	currently_drawing_blood = 0
+	update_icon()
+
+/obj/machinery/bot/bloodbot/attack_paw(mob/user as mob)
+	return attack_hand(user)
+
+/obj/machinery/bot/bloodbot/attack_hand(mob/user as mob)
+	if(..())
+		return
+	var/dat
+	dat += "<TT><B>Acula-class Blood Donation Bot v0.1</B></TT><BR><BR>"
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/datum/reagent/blood/B = H.get_blood(H.vessel)
+		if(B.data["virus2"])
+			dat += "<span class='danger'>WARNING: Viral agent detected. Ineligable for blood donation.</span><BR>"
+		else
+			dat += {"Welcome [H]! Your blood level is [round(B.volume/560*100)]%, and your blood type is [B.data["blood_type"]].<BR>
+				You must have at least [round(509/560*100)]% to donate blood. <a href='?src=\ref[src];donate=[1]'>Donate now!</a><BR>
+				The next reward will be dispensed after [55 - since_last_reward] units of blood are donated.<BR>"}
+	dat += {"Status: <A href='?src=\ref[src];power=1'>[src.on ? "On" : "Off"]</A><BR>"
+		Maintenance panel panel is [src.open ? "opened" : "closed"]<BR>"}
+	if(!src.locked || issilicon(user))
+		dat += "<TT>Blood Storage: "
+		var/counter = 0
+		if(!contents.len)
+			dat += "There are no blood packs available."
+		for (var/obj/item/weapon/reagent_containers/blood/E in contents)
+			counter++
+			dat += "Slot [counter]: [E] <A href='?src=\ref[src];slot=\ref[E]'>(Eject)</A><BR>"
+		dat += "The speaker switch is [src.quiet ? "off" : "on"]. <a href='?src=\ref[src];togglevoice=[1]'>Toggle</a><br>"
+
+	user << browse("<HEAD><TITLE>Dr. Acula Controls</TITLE></HEAD>[dat]", "window=autoblood")
+	onclose(user, "autoblood")
+	return
+
+/obj/machinery/bot/bloodbot/Topic(href, href_list)
+	if(..())
+		return
+	usr.set_machine(src)
+	add_fingerprint(usr)
+	if(href_list["power"] && allowed(usr))
+		if(on)
+			turn_off()
+		else
+			turn_on()
+
+	else if(href_list["slot"] && contents.len)
+		var/obj/item/weapon/reagent_containers/blood/E = locate(href_list["slot"])
```

Comment:
That's not the issue. Because this code is poorly sanitized you can teleport any object you want to yourself provided you somehow got note of the reference.

And getting a reference is not hard. You could even guess references because each base BYOND type is in a separate reference space.